Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -91,4 +179,151 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::symbol_species_getter)
     return vm.this_value();
 }
 
+// get RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#get-regexpinput
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_getter)
+{
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+
+    // 1. Return ? GetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]]).
+    return js_string(vm, regexp_constructor->legacy_static_properties().input());
```

Comment:
These are missing step 4:

```
4. If val is empty, throw a TypeError exception.
```

Can also be implemented as a standalone function by passing `&RegExpLegacyStaticProperties::input` as a function pointer, similar to this:
https://github.com/SerenityOS/serenity/blob/486775f9fe4fafb15df2d8fbe3276c796595eeec/Userland/Libraries/LibJS/Runtime/AbstractOperations.cpp#L377

Reply:
```c++
// DEFINE get_legacy_regexp_static_property
ThrowCompletionOr<Value> get_legacy_regexp_static_property(VM& vm, FunctionObject const& constructor, Value this_value, Function<Optional<String>(void)> const& property_getter);

JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_getter)
{
    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();

    // 1. Return ? GetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]]).
    Function<Optional<AK::String>()> const property_getter = [&] {
        return regexp_constructor->legacy_static_properties().input();
    };
    return TRY(get_legacy_regexp_static_property(vm, *regexp_constructor, vm.this_value(), property_getter));
}
```